### DISCUSSION TITLE
How does policy ordering work ?

### DISCUSSION THREAD

**Discussion by paulchabanon** [goauthentik/authentik#9358] [Q&A] [answered]
Hello,

I am currently testing Authentik on my server and I really like it!
Big props to all the team !

I have been doing a bit of testing and there is some fundamental logic that I seem to miss.
Here is my case:

I followed the instructions at the bottom of this page https://docs.goauthentik.io/integrations/sources/google/ to do the username mapping and it works like a charm but, I am wondering what is going on when they are multiple policies conditioning a stage.

I have here two policies with orders 0 and 1.
![2024-04-19_19h50m27s](https://github.com/goauthentik/authentik/assets/1833175/74e73a16-5486-405a-8e11-a64d52e0db82)

Here is the corresponding diagram:
![2024-04-19_19h49m49s](https://github.com/goauthentik/authentik/assets/1833175/0ecae239-b970-4b12-bc6e-a0c67d18312b)

I was expecting a "first matched" policy system, especially because I have selected "any" for the policy matching option in the stage binding, which I understand as: if the first policy doesn't match the second is tried, then the third, and so on.
But, in the diagram my policies appear on the same "level" (or hight) which I found confusing.
I don't really understand why that is and why they are both connected to the flow start.

My tests seam to show that, because default-source-enrollment-if-username is "order 1", it appears to be never evaluated.
In the diagram we can see that non of the outputs of dz-google-sso-username-mapping-policy are going back to default-source-enrollment-if-username. Is their even a scenario where default-source-enrollment-if-username is evaluated ?

Thank you in advance for those clarifications.

**Comment by aureateflux** [comment, marked as answer]
You've probably figured this out by now, but I wanted to reply to this because I found your post while I was trying to figure out a similar problem.

Your assumptions aren't that far off-- the two Policy Engine options change the behavior of the policy check. "Any" should execute the stage if any single bound, enabled policy matches, while "All" should execute the stage only if every bound, enabled policy matches.

The problem is that they're .. mislabeled. Currently (June 12, 2024), if you want the "any" behavior, you have to select "all," and vice versa.

Hope this helps!

**Reply by paulchabanon** [reply]
Thank you for those clarifications ;)
